modules/core/src/main/scala/org/scalasteward/core/vcs/github/authentication.scala

```diff
@@ -16,13 +16,61 @@
 
 package org.scalasteward.core.vcs.github
 
+import java.io.FileReader
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.{KeyFactory, PrivateKey, Security}
+import java.util.Date
+
+import better.files.File
 import cats.Applicative
 import cats.syntax.all._
+import io.jsonwebtoken.{Jwts, SignatureAlgorithm}
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.util.io.pem.PemReader
 import org.http4s.headers.Authorization
 import org.http4s.{BasicCredentials, Request}
 import org.scalasteward.core.vcs.data.AuthenticatedUser
+import scala.concurrent.duration.Duration
+import scala.util.Using
 
 object authentication {
   def addCredentials[F[_]: Applicative](user: AuthenticatedUser): Request[F] => F[Request[F]] =
     _.putHeaders(Authorization(BasicCredentials(user.login, user.accessToken))).pure[F]
+
+  Security.addProvider(new BouncyCastleProvider())
+
+  private[this] def parsePEMFile(pemFile: File): Array[Byte] =
+    Using.resource(new PemReader(new FileReader(pemFile.toJava))) { reader =>
+      reader.readPemObject().getContent
+    }
+
+  private[this] def getPrivateKey(keyBytes: Array[Byte]): PrivateKey = {
+    val kf = KeyFactory.getInstance("RSA")
+    val keySpec = new PKCS8EncodedKeySpec(keyBytes)
+    kf.generatePrivate(keySpec)
+  }
+
+  private[this] def readPrivateKey(file: File): PrivateKey = {
+    val bytes = parsePEMFile(file)
+    getPrivateKey(bytes)
+  }
+
+  /** [[https://docs.github.com/en/free-pro-team@latest/developers/apps/authenticating-with-github-apps#authenticating-as-a-github-app]] */
+  def createJWT(app: GitHubApp, ttl: Duration): String = {
```

Comment:
This should be suspended in `F[_]` since it contains side effects (reading the key file and getting the current time).

I guess the `ttl` could also be a `FiniteDuration` since infinite `Duration`s don't make much sense here.